syntax = "proto3";

package komputer;

message Komputer {
    int32 id = 1;
    string name = 2;
    string description = 3;
    int32 price = 4;
    string image_url = 6;
    int32 stock = 7;
}

message KomputerListRequest {
}

message KomputerListResponse {
    repeated Komputer komputers = 1;
}

message KomputerRequest {
    int32 id = 1;
}

message KomputerResponse {
    Komputer komputer = 1;
}

message KomputerCreateRequest {
    string name = 1;
    string description = 2;
    int32 price = 3;
    string image_url = 4;
    int32 stock = 5;
}

message KomputerCreateResponse {
    Komputer komputer = 1;
}

message KomputerUpdateRequest {
    int32 id = 1;
    string name = 2;
    string description = 3;
    int32 price = 4;
    string image_url = 5;
    int32 stock = 6;
}

message KomputerUpdateResponse {
    Komputer komputer = 1;
}

message KomputerDeleteRequest {
    int32 id = 1;
}

message KomputerDeleteResponse {
    string message = 1;
}

service KomputerService {
    rpc List(KomputerListRequest) returns (KomputerListResponse);
    rpc Get(KomputerRequest) returns (KomputerResponse);
    rpc Create(KomputerCreateRequest) returns (KomputerCreateResponse);
    rpc Update(KomputerUpdateRequest) returns (KomputerUpdateResponse);
    rpc Delete(KomputerDeleteRequest) returns (KomputerDeleteResponse);
}
